sw/device/silicon_creator/lib/drivers/keymgr.c

```diff
@@ -56,6 +56,9 @@ static rom_error_t expected_state_check(uint32_t expected_state) {
 
   // Read the working state with sec_mmio so that we can check the expected
   // value periodically.
+  // Note: Any checks in ROM_EXT after a keymgr transition must be performed
+  // after the value in the expectations table is updated using a
+  // `sec_mmio_read32()` call.
```

Comment:
nit: Maybe worth to just then add `sec_mmio_read32()` to the key advance functions. This can be done in a follow up.

Reply:
Yes, that is a good point.

Reply:
It looks like you already added this note to `keymgr_advance_state()` in e7a3c5a29a :)  I'll just add more details for other functions.